client/components/UserDecalPanel.js

```diff
@@ -87,6 +90,37 @@ class UserDetailsPanel extends Component {
     return null
   }
 
+  _renderDecalCode = (decal, index) => (
+    <>
+      <div className="decal-code" key={`code${index}`}>
+        <button
+          key={`toggle${index}`}
+          aria-label="toggle-decal"
+          className="icon decal-toggle"
+          onClick={this._handleToggle}
+          // title={maxNicksReached ? 'You\'ve used all your nicknames' : 'Add new nickname'}
+          type="button">
+          <FontAwesomeIcon icon={this.state.formOpen[index] ? 'eye-slash' : 'eye'} fixedWidth />
```

Comment:
again, avoid using index. Better to use the decal's id like we do with key attributes.